src/internal-util/fs.ts

```diff
@@ -11,6 +12,28 @@ export async function existsFile(path: string): Promise<boolean> {
   return !!stats && stats.isFile();
 }
 
+export async function existsSourceFile(path: string, filename: string = 'default'): Promise<string | undefined> {
```

Comment:
why not simply use `require.resolve` to resolve a module with any supported file extension?